frontend/app/src/components/home/P2PSwapContent.svelte

```diff
@@ -0,0 +1,316 @@
+        } else if (content.status.kind === "p2p_swap_reserved") {
+            if (acceptedByYou) {
+                instructionText = $_("p2pSwap.youReserved");
+            } else {
+                instructionText = $_("p2pSwap.reservedBy", {
+                    values: { user: `@UserId(${content.status.reservedBy})` },
+                });
+            }
+            buttonText = i18nKey("p2pSwap.reserved");
+        } else if (content.status.kind === "p2p_swap_accepted") {
+            if (acceptedByYou) {
+                instructionText = $_("p2pSwap.youAccepted");
+            } else {
+                instructionText = $_("p2pSwap.acceptedBy", {
+                    values: { user: `@UserId(${content.status.acceptedBy})` },
+                });
+            }
+            buttonText = i18nKey("p2pSwap.accepted");
+        } else if (content.status.kind === "p2p_swap_completed") {
+            if (acceptedByYou) {
+                instructionText = $_("p2pSwap.youCompleted");
+            } else {
+                instructionText = $_("p2pSwap.completed", {
+                    values: { user: `@UserId(${content.status.acceptedBy})` },
+                });
+            }
+            buttonText = i18nKey("p2pSwap.accepted");
+        }
+
+        summaryText = i18nKey("p2pSwap.summary", {
+            fromAmount,
+            toAmount,
+            fromToken: content.token0.symbol,
+            toToken: content.token1.symbol,
+        });
+    }
+
+    function onAcceptOrCancel(e: MouseEvent) {
+        if (e.isTrusted && !buttonDisabled) {
+            confirming = true;
+        }
+    }
+
+    function cancel(yes: boolean): Promise<void> {
+        confirming = false;
+
+        if (yes && me) {
+            client
+                .cancelP2PSwap(
+                    messageContext.chatId,
+                    messageContext.threadRootMessageIndex,
+                    messageId,
+                )
+                .then((resp) => {
+                    if (resp.kind !== "success") {
+                        showFailureToast(resp);
+                    }
+                });
+        }
+
+        return Promise.resolve();
+    }
+
+    function accept() {
+        confirming = false;
+
+        if (!me) {
+            client
+                .acceptP2PSwap(
+                    messageContext.chatId,
+                    messageContext.threadRootMessageIndex,
+                    messageId,
+                )
+                .then((resp) => {
+                    if (resp.kind !== "success") {
+                        showFailureToast(resp);
+                    }
+                });
+        }
+    }
+
+    function showFailureToast(response: AcceptP2PSwapResponse | CancelP2PSwapResponse) {
+        let key: string = response.kind;
+
+        if (key === "already_reserved" || "already_completed") {
+            key = "already_accepted";
+        }
+
+        toastStore.showFailureToast(i18nKey("p2pSwap." + key));
```

Comment:
Both accept and cancel can return kind of `internal_error` which is not present in the i18n files.